src/qtgui/plotter.cpp

```diff
@@ -1168,7 +1159,7 @@ void CPlotter::draw(bool newData)
     const qint32 maxbin = std::min(endBin + 1, m_fftDataSize - 1);
 
     const qint32 xmin = qRound((double)(minbin - startBin) * xScale);
-    const qint32 xmax = qRound((double)(maxbin - startBin) * xScale);
+    const qint32 xmax = std::min(qRound((double)(maxbin - startBin) * xScale), qRound(w));
```

Comment:
We have to be more careful here to avoid drawing outside the bounds of the waterfall image.